core/sql/executor/ex_hash_grby.cpp

```diff
@@ -452,6 +463,8 @@ short ex_hash_grby_tcb::work() {
             hasFreeTupp_ = TRUE;
     	};
       };
+      if (bmoStats_)
+         bmoStats_->setBmoPhase(PHASE_END-HGB_READ_PHASE);
```

Comment:
This is executed for every row. Would it be cheaper to do this in the two places where we set state_ to HASH_GRBY_READ_CHILD?
Another question not about correctness but more related to personal taste: Why not simply assign values 1, 2, 3 etc. in the enum instead of subtracting two enums here? Feel free to ignore if you like this solution better.